Auth

julie1013 opened the pull request:
Set up the event handlers/ajax requests/ui success and fail handlers for sign up, sign in, and change password.

julie1013 committed:
Created events.js, api.js, ui.js files

Created events.js, api.js, ui.js files

julie1013 committed:
Set up sign-up, sign-in, and change password

Set up event handlers, ajax requests, and UI success/failure handlers
for sign-up, sign-in, and change password.

kopius commented:
Just pulled these changes down to a branch and tested them. Looks great. 

1 fix: please change app.host to localhost:3000. Then it should work!

Eventually we should refactors events.js, api.js, and ui.js into an `auth` folder to separate these concerns from the file upload concerns, but this is not urgent.

julie1013 committed:
Moved events, api, and ui file; changed backend host

Moved events, api, and ui files into a folder called "auth".
Fixed backend host in app, setting it to localhost:3000 as it should be.
